src/Http/Routing/src/Builder/MinimalActionEndpointRouteBuilderExtensions.cs

```diff
@@ -184,6 +185,14 @@ public static class MinimalActionEndpointRouteBuilderExtensions
             // Add MethodInfo as metadata to assist with OpenAPI generation for the endpoint.
             builder.Metadata.Add(action.Method);
 
+            // Methods defined in a top-level program are generated as statics so the delegate
+            // target will be null. Inline lambdas are compiler generated properties so they can
+            // be filtered that way.
+            if (action.Target == null || !TypeHelper.IsCompilerGenerated(action.Method.Name))
+            {
+                builder.Metadata.Add(new EndpointNameMetadata(action.Method.Name));
```

Comment:
Check the MVC implementation, I believe this can be problematic since the endpoint name needs to be globally unique

Reply:
Yeah, I skimmed through the MVC implementation and found this:

https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs#L329-L341

Which addresses the equivalent scenario of something like this in minimal APIs:

```
builder.MapPost("/foo", SomeMethod);
builder.MapGet("/foobar", SomeMethod);
```

We can use the HTTP method to disambiguate between the two in that case. Other than that, I couldn't find any places where we check for global uniqueness.

Reply:
Ah, just found https://github.com/dotnet/aspnetcore/blob/main/src/Http/Routing/src/EndpointNameAddressScheme.cs#L38.

So this is actual a problem that we have sans the automagic addition of endpoint names here.

We don't currently do anything to stop users from doing:

```
app.MapGet("/foo", () => {}).WitName("SomeName");
app.MapGet("/bar", () => {}).WitName("SomeName");
```

Reply:
We had an out-of-band discussion about this. Turns out this is an issue with minimal APIs, but not for any reasons related to this change set. Tracking this:

https://github.com/dotnet/aspnetcore/issues/35436